### PR TITLE
Simplify glyph function creation

### LINKED ISSUE #1917: remove _glyphfunctions
With the old implicit glyph functions gone, the reason for this complicated machinery is gone too. These should be made simple, normal methods on `plotting.Figure`.

### PR DESCRIPTION
**Pull request by bryevdv**
issues: closes #1917 

* configure glyphs with a suggested argument order
* automate glyph function docstring creation
* move all functionality to plotting.py and plotting_helpers.py

### CODE REVIEW

**Comment by bryevdv** [comment]
@bokeh/dev This can still be improved I'm sure. The auto-generated reference docs are already much better. There is a test deploy here:

http://bokeh.pydata.org/en/test/docs/reference/plotting.html

Would appreciate suggestions on improvements to the presentation of these reference docs for glyph functions as well as anything else.

**Comment by bryevdv** [comment]
In particular, are there more useful types/descriptions to provide for the params.

**Comment by bryevdv** [comment]
@bokeh/dev request review

File: bokeh/models/glyphs.py [hunk 1]
```diff
@@ -462,12 +493,12 @@ class ImageURL(Glyph):
     """)
 
     angle = AngleSpec(default=0, help="""
-    The angles to rotate the images, in radians as measured from the
-    horizontal.
+    The angles to rotate the images, as measured from the horizontal.
```
Comment (almarklein) [inline bokeh/models/glyphs.py]: Why remove the mentioning of radians? Seems like useful info.

**Comment by almarklein** [comment]
Some suggestions from browsing through that HTML doc:

* In the docs for `annular_wedge`, the `Enum` prop (for 'direction') is a link, which can be convenient, though perhaps distracting. All the other prop classes are plain text.
* The examples section for the annulus seems indented into the return section.
* The example plots are good, let's do more of those (e.g. I have no idea what an annulus looks like). However, the code for all examples is almost the same, maybe hide it to make it more concise? Also hide toolbar; I would just want to see what that glyph looks like.
* quadratic might need more docs: is this a glyph to create quads or a quadratic spline? If the former, x0, x1, y0, y1, should probably be called corner coordinates or something.
* Maybe a TOC (list of glyphs with links to their docs) at the top of the page?

**Comment by bryevdv** [comment]
I have fixed the example indention. I personally think leaving the code is important, though if more people have opinions I'd like to hear them. 

`quadratic` creates a quadratic spline. There is a separate `quad` glyph for axis-aligned rectangles. I can definitely believe there are better names for the coordinates (and for the cubic `bezier` glyph), I just used what I had encountered previously. 

TOC is a very nice idea, but it can go in a different PR I think. (Auto-generating that will require more sphinx tooling than I'd like to include in this PR, which is really about getting rid of/simplifying  `_glyphfunctions.py`)

@bokeh/dev any other comments?

File: bokeh/models/glyphs.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/models/glyphs.py]: because now the units are configurable and so that doc was incorrect/misleading as stated.

**Comment by canavandl** [comment]
For some reason, the `height_units` prop isn't appearing in the `rect` method docstring (`width_units` does). Let me find out why/see if any other props are missing before merging.

File: bokeh/models/glyphs.py (hunk 1 above)
Comment (bryevdv) [inline bokeh/models/glyphs.py]: Specifically:

screen shot 2015-10-05 at 12 29 09 pm

**Pull request merged**